### PR TITLE
glfs: implement async read/write/flush versions

### PR DESCRIPTION
**Pull request by pkalever**
Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

### CODE REVIEW

**Comment by pkalever** [comment]
@mikechristie @lxbsz please-review

File: glfs.c [hunk 1]
```diff
@@ -539,54 +550,122 @@ static void tcmu_glfs_close(struct tcmu_device *dev)
...
 
-	ret = glfs_pwritev(state->gfd, iov, iov_cnt, offset, ALLOWED_BSOFLAGS);
-	if (ret != length) {
-		ret =  tcmu_set_sense_data(cmd->sense_buf, MEDIUM_ERROR,
-					   ASC_WRITE_ERROR, NULL);
-	} else {
-		ret = SAM_STAT_GOOD;
+	cookie = calloc(1, sizeof(*cookie));
+	if (!cookie) {
+		tcmu_err("Could not allocate cookie: %m\n");
```
Comment (lxbsz) [inline glfs.c]: I think as Mike mentioned before in my some pull requests, we'd better convert to use tcmu_dev_XXX() instead.

The same with the other places.

File: glfs.c [hunk 2]
```diff
@@ -539,54 +550,122 @@ static void tcmu_glfs_close(struct tcmu_device *dev)
...
-			   struct iovec *iov, size_t iov_cnt, size_t length,
-			   off_t offset)
+static int tcmu_glfs_write(struct tcmu_device *dev,
+                           struct tcmulib_cmd *cmd,
+                           struct iovec *iov, size_t iov_cnt,
+                           size_t length, off_t offset)
 {
 	struct glfs_state *state = tcmu_get_dev_private(dev);
-	ssize_t ret;
+	glfs_cbk_cookie *cookie = NULL;
```
Comment (lxbsz) [inline glfs.c]: Maybe no need to initialize to NULL here.

File: glfs.c (hunk 1 above)
Comment (pkalever) [inline glfs.c]: Xiubo Lee, you are right, and that is in my mind too, but I would like to make that as part of an another patch, as it needs changes all over the file and may not be related to the subject of the patch.

So for now, kept it similar with other places.

File: glfs.c (hunk 2 above)
Comment (pkalever) [inline glfs.c]: Yeah, right!
The glitch was due to some last minute restructure, will send an another version.

Thanks!

File: glfs.c (hunk 1 above)
Comment (lxbsz) [inline glfs.c]: Yes, that sounds nice :)
